### ISSUE TITLE
Refactor automatic A11y testing with AXE-Core in Storefront

### ISSUE THREAD

**Issue by o-hook** [porsche-design-system/porsche-design-system#1892]
Scope
---
To shift-left accessibility testing to fetch possible issues earlier in the development phase, we have to refactor automatic a11y testings of our Storefront pages.

Out of Scope
---
- UX Platform
- Component based testing

Notes  
---
- Find a solution how to deal with found accessibility issues and how strict deployment should be (e.g. in which case should the deployment be stopped in regards of the severity of issues - critical, serious, moderate and minor)
- AXE-Core Testing Engine / Rulesets: https://github.com/dequelabs/axe-core 
- AXE-Core packages: https://github.com/dequelabs/axe-core-npm 
- AXE-Core API docs: https://github.com/dequelabs/axe-core/blob/master/doc/API.md
- An a11y test is already integrated, see `axe.e2e.ts` and `navigation.e2e.ts`
- Sitemap should be generated once by `dead-link-checker.e2e.ts` and/or `accessibility.e2e.ts`
- Currently we use `wcag2aa` but should be `wcag21aa`

Acceptance criteria
---
- [x] For all pages of Storefront any a11y errors are fixed
- [x] A11y testing is separated in dedicated spec file
- [x] A11y testing loops through a generated sitemap array of e.g. `dead-link-checker.e2e.ts` to catch all Storefront pages
- [x] The whole page is reflected in a11y test (including Sidebar/Nav even if it seems repetitive)
- [x] A11y testing reflects Playground in light/dark mode and Vanilla JS, React, Angular implementation variant for each page
- [x] Optimize error output in test execution
- [x] A11y presets are identical to Axe Monitor (`wcag21aa`)

Subtasks
---
- [x] create sitemap.json fixture
- [x] separate a11y tests from dead link checker test
